etcdserver/etcdhttp/client_security.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdhttp
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/coreos/etcd/etcdserver"
+	"github.com/coreos/etcd/etcdserver/etcdhttp/httptypes"
+	"github.com/coreos/etcd/etcdserver/security"
+)
+
+type securityHandler struct {
+	store       *security.Store
+	clusterInfo etcdserver.ClusterInfo
+}
+
+func handleSecurity(mux *http.ServeMux, server *etcdserver.EtcdServer) {
+	sec := securityHandler{
+		store:       security.NewStore(server, defaultServerTimeout),
+		clusterInfo: server.Cluster,
+	}
+	mux.HandleFunc(securityPrefix+"/roles", sec.baseRoles)
+	mux.HandleFunc(securityPrefix+"/roles/", sec.handleRoles)
+	mux.HandleFunc(securityPrefix+"/users", sec.baseUsers)
+	mux.HandleFunc(securityPrefix+"/users/", sec.handleUsers)
+}
+
+func (sec *securityHandler) handleRoles(w http.ResponseWriter, r *http.Request) {
```

Comment:
it would be great if you can switch the order of handleRoles and baseRoles, since it is consistent with the line39 and line 40. also the forRole and actionRole function will be closer to handleRoles which can improve readability.